java/client/src/org/openqa/selenium/remote/internal/OkHttpClient.java

```diff
@@ -116,11 +117,20 @@ public void close() {
 
     @Override
     public HttpClient createClient(URL url) {
-      okhttp3.OkHttpClient client = new okhttp3.OkHttpClient.Builder()
+      okhttp3.OkHttpClient.Builder builder = new okhttp3.OkHttpClient.Builder()
           .connectionPool(pool)
           .followRedirects(true)
-          .followSslRedirects(true)
-          .build();
+          .followSslRedirects(true);
+
+      String urlUserInfo = url.getUserInfo();
+      if (urlUserInfo != null) {
+        String[] userInfo = urlUserInfo.split(":");
+        String basic = Credentials.basic(userInfo[0], userInfo[1]);
```

Comment:
We need a bounds check here. If `urlUserInfo` lacks a `:` we'll only have `userInfo[0]`